contracts/extension/GovernorVotesSuperQuorumFractionUpgradeable.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: MIT
+// Original design by OpenZeppelin
+// Modified by Dennison Bertram and Arthur Abeilice  @ Tally.xyz
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/governance/extensions/GovernorVotes.sol";
+import "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import "@openzeppelin/contracts/utils/structs/Checkpoints.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+/// @title GovernorVotesSuperQuorumFractionUpgradeable
+/// @notice Extends GovernorVotes to add super quorum functionality based on a fraction of the total token supply.
+/// @dev This contract adds an additional quorum mechanism to the standard GovernorVotes functionality.
+abstract contract GovernorVotesSuperQuorumFractionUpgradeable is Initializable, GovernorVotes {
+    using Checkpoints for Checkpoints.Trace208;
+
+    Checkpoints.Trace208 private _superQuorumNumeratorHistory;
+
+    event SuperQuorumNumeratorUpdated(uint256 oldQuorumNumerator, uint256 newQuorumNumerator);
+
+    /// @dev Thrown when the provided super quorum numerator is not valid.
+    error GovernorInvalidSuperQuorumFraction(uint256 quorumNumerator, uint256 superQuorumDenominator);
+
+    /// @notice Initializes the contract with a super quorum numerator.
+    /// @param quorumNumeratorValue The initial super quorum numerator.
+    constructor(uint32 quorumNumeratorValue) {
```

Comment:
@afa7789 there should be no constructors in any code that can be deployed as a minimal proxy. (IE: this should be replaced with [GovernorVotes-Upgradable](https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/master/contracts/governance/extensions/GovernorVotesUpgradeable.sol)